com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoopDef.hlsl

```diff
@@ -316,7 +335,7 @@ EntityLookupParameters InitializeEntityLookup(uint tile, uint2 zBinRange, uint c
 }
 
 // Internal. Do not call directly.
-uint TryFindEntityIndex(uint i, inout EntityLookupParameters params, out uint entityIndex)
+uint TryFindEntityIndex(uint i, inout EntityLookupParameters params, out uint entityIndex, bool depthSorted)
```

Comment:
Please move "depthSorted" into EntityLookupParameters. Do not pollute the API.

Reply:
I thought the idea here was that if we pass constant 'true' or 'false' as parameters the compiler will be able to optimize out the branch.

Reply:
Either way should be optimizable.

Reply:
Whatever we do it need to be know at compile time to benefit from compilation optimization (but true that we should measure the impact, keeping it dynamic would be interesting :) )

Reply:
Guess we should measure it. I will expect that static compilation is the way to go. But if we don't measure any benefit with dynamic condition, then it is better to have it dynamic.

Reply:
I think what @EvgeniiG means is that compiler should be clever enough to optimize it even if it is part of the parameters structure.